lib/utils.js

```diff
@@ -34,27 +34,27 @@ function relayEvents(listener, emitter, events) {
   events.forEach(eventName => listener.on(eventName, event => emitter.emit(eventName, event)));
 }
 
-// Grab Kerberos values if they exist, otherwise set them to null
-let Kerberos = null;
-let MongoAuthProcess = null;
+function retrieveKerberos() {
+  let kerberos;
 
-try {
-  const kerberos = requireOptional('kerberos');
-  if (kerberos) {
-    Kerberos = kerberos.Kerberos;
-    MongoAuthProcess = kerberos.processes.MongoAuthProcess;
+  try {
+    kerberos = requireOptional('kerberos');
+  } catch (err) {
+    if (err.code !== 'MODULE_NOT_FOUND') {
```

Comment:
If we are now deferring requiring `kerberos` until we actually auth with it, then we shouldn't swallow `MODULE_NOT_FOUND`. If the module isn't found, auth should error.

Reply:
We're swallowing it to control the text of the returned error to be something more user friendly than "module not found"